### ISSUE TITLE
let-else in a loop flagged as unreachable code

### ISSUE THREAD

**Issue by josephg** [intellij-rust/intellij-rust#10104]
## Environment

* **IntelliJ Rust plugin version:** 0.4.187.5175-223
* **Rust toolchain version:** 1.67.0 (fc594f156 2023-01-24) aarch64-apple-darwin
* **IDE name and version:** CLion 2022.3.2 (CL-223.8617.54)
* **Operating system:** macOS 13.0.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc

## Problem description

The following code is flagged as unreachable, when it shouldn't be. The problem seems to be in some interaction between let-else and the loop.

![image](https://user-images.githubusercontent.com/47413/218003589-2bf43cba-05e3-44fc-8b0b-b3fdbd5f7f16.png)

## Steps to reproduce
```rust
#[test]
fn asdf() {
    loop {
        let r = Ok(123);
        let Ok(msg) = r else { break; };
        println!("I will be executed forever!");
    }
}
```
